build.gradle

```diff
@@ -2286,8 +2286,9 @@ project(':streams:streams-scala') {
     api project(':streams')
 
     api libs.scalaLibrary
-    api libs.scalaCollectionCompat
-
+    if ( versions.baseScala == '2.12' ) {
+      api libs.scalaCollectionCompat
```

Comment:
Can you please add a comment here explaining the reason for this removal of dependency, basically, capture what you wrote in PR here. It helps the readers of the code understand why certain actions were taken.